video-app/src/app/core/youtube/youtube.service.ts

```diff
@@ -1,52 +1,56 @@
-import { Injectable} from '@angular/core';
+import { Injectable } from '@angular/core';
 import { Observable, of } from 'rxjs';
 import { switchMap } from 'rxjs/operators';
 import { HttpClient } from '@angular/common/http';
-import { Video, Passwords, Source  } from '../models';
-import { Router } from '@angular/router';
-import { StreamingPlatformService } from '../common/streamingPlatform.service';
+import { Video, Passwords } from '@core/models';
+import { StreamingPlatformService } from '@core/common/streamingPlatform.service';
+import { PlatformEnum } from '@core/enums/platform.enum';
+import { AddVideoResult } from '@core/models/addVideoResult';
 
 @Injectable()
 export class YoutubeService {
 
   public passwords = new Passwords();
 
-  constructor( private readonly http: HttpClient
-    ,          private readonly router: Router
-    ,          private readonly streamingPlatform: StreamingPlatformService ) {}
+  constructor(private readonly http: HttpClient
+    ,         private readonly streamingPlatformService: StreamingPlatformService) { }
 
-  public getVideo(videoId: string): Observable<any> {
+  public addVideo(videoId: string): Observable<AddVideoResult> {
 
     return this.http.get('https://www.googleapis.com/youtube/v3/videos?id='
       + videoId +
       '&key='
       + this.passwords.YouTube +
       '&part=snippet,contentDetails,statistics,status')
-        .pipe(
-          switchMap((response: any) => {
-
-            if (response.items.length === 0) {
-              return of(true);
-            }
-
-            const item = response.items[0];
-
-            const video = new Video();
-            video.sourceType = Source.Youtube;
-            video.source = 'Youtube';
-            video.id = item.id;
-            video.title = item.snippet.title;
-            video.date = item.snippet.publishedAt;
-            video.image = item.snippet.thumbnails.default.url;
-            video.playesCount = item.statistics.viewCount;
-            video.playsCountDescription = item.statistics.viewCount;
-            video.likesCount = item.statistics.likeCount;
-            video.urlCode = videoId;
-            video.creationDate = new Date();
-
-            this.streamingPlatform.saveVideo(video);
-            return of(false);
-          })
-        );
+      .pipe(
+        switchMap((response: any) => {
+        const videoResult = new AddVideoResult();

```

Comment:
wszystko co się dzieje w tym switchMapu, czyli całe parsowanie na nasz obiekt Video, wydziel do osobnej metody i najlepiej odpal ja dopiero w effectcie, przegadamy to na callu:)